helios-client/src/main/java/com/spotify/helios/common/descriptors/HostSelector.java

```diff
@@ -49,6 +49,30 @@ public boolean test(final String a, final Object b) {
       public boolean test(final String a, final Object b) {
         return !Objects.equals(a, b);
       }
+    }),
+    IN("in", new BiPredicate<String, Object>() {
+      @Override
+      public boolean test(final String a, final Object b) {
+        final String[] parts = ((String) b).replaceAll("(\\(|\\)| )", "").split(",");
+        for (final String part : parts) {
+          if (part.equals(a)) {
+            return true;
+          }
+        }
+        return false;
+      }
+    }),
+    NOT_IN("notin", new BiPredicate<String, Object>() {
+      @Override
+      public boolean test(final String a, final Object b) {
+        final String[] parts = ((String) b).replaceAll("(\\(|\\)| )", "").split(",");
+        for (final String part : parts) {
+          if (part.equals(a)) {
+            return false;
+          }
+        }
+        return true;
+      }
```

Comment:
This looks to be the same logic except the boolean values are reversed. How about making re-using the IN definition to avoid duplication?

```java
NOT_IN("notin", new BiPredicate<String, Object>() {
  @Override
  public boolean test(final String a, final Object b) {
    return !IN.test(a, b);
  }
}
```